services/app-web/src/pages/StateSubmission/StateSubmissionForm.tsx

```diff
@@ -165,7 +170,24 @@ export const StateSubmissionForm = (): React.ReactElement => {
         }
     }
 
-    // Set up side effects
+    //Delete file from S3 if file has not been submitted
+    const handleDeleteFile = async (key: string) => {
+        const isSubmittedFile =
+            previousDocuments &&
+            Boolean(
+                previousDocuments.some((previousKey) => previousKey === key)
+            )
+
+        if (!isSubmittedFile) {
+            const result = await deleteFile(key)
+            if (isS3Error(result)) {
+                throw new Error(`Error in S3 key: ${key}`)
```

Comment:
what happens if we throw here? I think we’d want to display an error allowing for the deletion to be re-tried or something?

Reply:
@JasonLin0991 if you have capacity, could you add a quick storybook story for this case ^ to FileUpload. We have a bunch of examples already in there to make it easier to verify these kinds of edge cases about FileUpload. We already have helpers to mock deleting scanning and uploading. Looks like we are missing a delete cases. 

If you can't get to it ping me, I can add on to one of my PRs this week.